sigmastate/src/main/scala/sigmastate/serialization/FuncValueSerializer.scala

```diff
@@ -28,9 +28,14 @@ case class FuncValueSerializer(cons: (IndexedSeq[(Int, SType)], Value[SType]) =>
 
   override def parse(r: SigmaByteReader): Value[SType] = {
     val argsSize = r.getUIntExact
+    // NO-FORK: in v5.x getUIntExact may throw Int overflow exception
+    // in v4.x r.getUInt().toInt is used and may return negative Int instead of the overflow
+    // in which case the array allocation will throw NegativeArraySizeException
     val args = safeNewArray[(Int, SType)](argsSize)
     cfor(0)(_ < argsSize, _ + 1) { i =>
-      val id = r.getUIntExact
+      val id = r.getUInt().toInt
+      // Note, when id < 0 as a result of Int overflow, the r.valDefTypeStore(id) won't throw
+      // but this will likely fail elsewhere
```

Comment:
Such comment is confusing, "likely fail elsewhere" does not really says anything, please specify how a transaction where a script has negative id encoded will fail. It it will fail, no need to revert getUIntExact

Reply:
clarified in comment